integration/spark/shared/src/test/java/io/openlineage/spark/agent/util/PathUtilsTest.java

```diff
@@ -162,10 +162,34 @@ void testFromCatalogTableWithStorage() throws URISyntaxException {
     assertThat(di.getSymlinks().get(0).getNamespace()).isEqualTo("hive://10.1.0.1:9083");
   }
 
+  @Test
+  void testFromCatalogTableWithGlue() throws URISyntaxException {
+    sparkConf.set(
+        "spark.hadoop.hive.metastore.client.factory.class",
+        "com.amazonaws.glue.catalog.metastore.AWSGlueDataCatalogHiveClientFactory");
+    sparkConf.set("spark.sql.catalogImplementation", "hive");
+    when(sparkContext.getConf()).thenReturn(sparkConf);
+    when(sparkSession.sparkContext()).thenReturn(sparkContext);
+
+    when(catalogTable.provider()).thenReturn(Option.apply("hive"));
+    when(catalogTable.storage()).thenReturn(catalogStorageFormat);
+    when(catalogTable.identifier()).thenReturn(TableIdentifier.apply(TABLE));
+    when(catalogStorageFormat.locationUri())
+        .thenReturn(Option.apply(new URI("s3://bucket/directory")));
+
+    DatasetIdentifier di = PathUtils.fromCatalogTable(catalogTable, Optional.of(sparkConf));
+    assertThat(di.getName()).isEqualTo("directory");
+    assertThat(di.getNamespace()).isEqualTo("s3://bucket");
+    assertThat(di.getSymlinks()).hasSize(1);
+    assertThat(di.getSymlinks().get(0).getName()).isEqualTo(TABLE);
+    assertThat(di.getSymlinks().get(0).getNamespace()).isEqualTo("glue");
```

Comment:
The notion of symlink is used to provide an information that two datasets are the same if any of the symlinks are equal. That's why symlinks should provide the same level of uniqueness like primary dataset names. Does this still hold for the solution proposed here? 

Is it possible to have two different glues (projects, namespaces, whatever naming they use) and table with the same name in each of them? If it is possible, we end up with different datasets being marked as the same dataset.

Reply:
The ARN(Amazon Resource Name) for Glue is documented [here](https://docs.aws.amazon.com/glue/latest/dg/glue-specifying-resource-arns.html#data-catalog-resource-arns), according to this [docs](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html) the `Amazon Resource Names (ARNs) uniquely identify AWS resources`.

So, probably, for namespace we could go for `aws:glue:<region>:<account-id>` or something similar?

Reply:
Using ARNs for purely-AWS resources would probably be a good idea.